client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -168,14 +240,28 @@ function mediaButton( editor ) {
 				return;
 			}
 
+			const media = MediaStore.get( selectedSite.ID, current.media.ID );
+			let mediaHasCaption = false;
+			let captionNode = null;
+
+			if ( media.isDirty ) {
+				current.media.transient = true;
+
+				captionNode = editor.dom.getParent( img, '.mceTemp' );
```

Comment:
The caption hack: if we want to keep captions sizing and alignment in sync with the updated images, we need to remove and add new caption shortcode on each image edit.